packages/property-provider/src/CredentialsProviderError.ts

```diff
@@ -0,0 +1,14 @@
+import { ProviderError } from "./ProviderError";
+
+/**
+ * An error representing a failure of an individual credential provider.
+ *
+ * This error class has special meaning to the {@link chain} method. If a
```

Comment:
Can you confirm this will link correctly in the doc? It might not since tsc doesn't have this reference in scope. If not let's use more descriptive phrases

Reply:
> Can you confirm this will link correctly in the doc?

Should this be out of scope of this PR?

Reply:
Created an issue for follow-up at https://github.com/aws/aws-sdk-js-v3/issues/3439